core/src/local/persistent_state.rs

```diff
@@ -410,28 +406,29 @@ impl PersistentState {
     // with exactly those values. I think the regular state implementation supports inserting
     // something like an Int and retrieving with a BigInt.
     fn insert(&mut self, batch: &mut WriteBatch, r: &[DataType]) {
-        for index in self.indices.iter_mut() {
+        for index in self.indices[1..].iter_mut() {
```

Comment:
Don't you need to increment `seq` for the first index if it's not a primary key too?